### ISSUE TITLE
signUp method of SignUp class from aws-amplify-react return void

### ISSUE THREAD

**Issue by NastyZ98** [aws-amplify/amplify-js#5340]
**Describe the bug**
Hi, I realized by extending the SignUp class that his `signUp()` method was not returning a promise ([source](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignUp.tsx#L206)), compared to the `SignIn()` method of the SignIn class, which return a promise ([source](https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignIn.tsx#L78)).
In my case I wanted to extends those classes to create a custom components.

Where I encountered difficulties was when I tried to disable the sign up button depending on the request made by the parent class (in this case SignUp class), because the signUp method does not return any promise.
It was made possible to disable the signIn button on my child component, because the `signIn()` method of the parent component (SignIn class) return a promise.

**To Reproduce**
```ts
import { SignIn } from "aws-amplify-react";

class CustomSignIn extends SignIn {
  constructor(props: any) {
    super(props);
    this._validAuthStates = ["signIn", "signedOut", "signedUp"];
  }
  showComponent(theme: any) {
      // We can't know when the request is resolved
       return (<Button onClick={() => super.signUp()}>Sign Up</Button>);
  }
}
```
**Expected behavior**
Make signUp method return a promise

I opened a pull request to make the signUp method return a promise.

<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 296.80 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.1 - /usr/local/bin/node
    Yarn: 1.22.4 - ~/Documents/Workspace/aws-boilerplate-cognito/node_modules/.bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 80.0.3987.149
    Safari: 13.1
  npmPackages:
    @bahmutov/add-typescript-to-cypress: ^2.1.2 => 2.1.2 
    @testing-library/jest-dom: ^4.2.4 => 4.2.4 
    @testing-library/react: ^9.3.2 => 9.5.0 
    @testing-library/user-event: ^7.1.2 => 7.2.1 
    @types/jest: ^24.0.0 => 24.9.1 
    @types/node: ^12.0.0 => 12.12.34 
    @types/react: ^16.9.0 => 16.9.31 
    @types/react-dom: ^16.9.0 => 16.9.6 
    @types/react-router-dom: ^5.1.3 => 5.1.3 
    @types/styled-components: ^5.0.1 => 5.0.1 
    add: ^2.0.6 => 2.0.6 
    aws-amplify: ^3.0.5 => 3.0.5 
    aws-amplify-react: ^4.1.4 => 4.1.4 
    cypress: ^4.3.0 => 4.3.0 
    global: ^4.4.0 => 4.4.0 
    react: ^16.13.1 => 16.13.1 
    react-dom: ^16.13.1 => 16.13.1 
    react-router-dom: ^5.1.2 => 5.1.2 
    react-scripts: 3.4.1 => 3.4.1 
    styled-components: ^5.0.1 => 5.0.1 
    typescript: ^3.8.3 => 3.8.3 
    webpack: ^4.42.1 => 4.42.1 
    yarn: ^1.22.4 => 1.22.4 
  npmGlobalPackages:
    npm: 6.13.4
```

</details>

**Comment by NastyZ98** [comment]
Hi,
I found a workaround by getting the parent state (`loading` for the signIn component and `requestPending` for the signUp component).
The fact that the state of signIn component has been initialized with an empty object confused me. This is why when I logged the state in the child constructor I was getting an empty object.
```ts
this.state = {};
```
https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignIn.tsx#L59

In the SignUp class component the state has been initialized with properties. I think it is a good practice to initialize the state with default properties before to use it, with this practice we will have the default values in the component that inherits
```ts
this.state = { requestPending: false };
```
https://github.com/aws-amplify/amplify-js/blob/master/packages/aws-amplify-react/src/Auth/SignUp.tsx#L67
